File: 1.21.9/src/main/java/com/bleudev/nine_lifes/client/custom/entity/model/WanderingArmorStandModel.java
package com.bleudev.nine_lifes.client.custom.entity.model;

import net.minecraft.client.model.ModelPart;
import net.minecraft.client.render.entity.model.ArmorStandEntityModel;
import net.minecraft.client.render.entity.state.ArmorStandEntityRenderState;
import net.minecraft.util.math.MathHelper;

public class WanderingArmorStandModel extends ArmorStandEntityModel {
    public WanderingArmorStandModel(ModelPart modelPart) {
        super(modelPart);
    }

    public void setAngles(ArmorStandEntityRenderState state) {
        super.setAngles(state);

        float f = state.limbSwingAnimationProgress;
        float g = state.limbSwingAmplitude;
        this.rightLeg.pitch = MathHelper.cos(f * 0.6662F) * 1.4F * g;
        this.leftLeg.pitch = MathHelper.cos(f * 0.6662F + (float)Math.PI) * 1.4F * g;
        this.rightArm.pitch = MathHelper.cos(f * 0.6662F + (float)Math.PI) * 1.4F * g;
        this.leftArm.pitch = MathHelper.cos(f * 0.6662F) * 1.4F * g;
    }
}

File: 1.21.10/src/main/kotlin/com/bleudev/nine_lifes/client/custom/entity/model/WanderingArmorStandModel.kt
package com.bleudev.nine_lifes.client.custom.entity.model

import net.minecraft.client.model.ArmorStandModel
import net.minecraft.client.model.geom.ModelPart
import net.minecraft.client.renderer.entity.state.ArmorStandRenderState
import kotlin.math.cos

class WanderingArmorStandModel(modelPart: ModelPart) : ArmorStandModel(modelPart) {
    override fun setupAnim(state: ArmorStandRenderState) {
        super.setupAnim(state)

        val f = state.walkAnimationPos
        val g = state.walkAnimationSpeed
        this.rightLeg.xRot = cos(f * 0.6662f) * 1.4f * g
        this.leftLeg.xRot = cos(f * 0.6662f + Math.PI.toFloat()) * 1.4f * g
        this.rightArm.xRot = cos(f * 0.6662f + Math.PI.toFloat()) * 1.4f * g
        this.leftArm.xRot = cos(f * 0.6662f) * 1.4f * g
    }
}